definitions/procedures/backup/snapshot/mount_foreman_db.rb

```diff
@@ -0,0 +1,46 @@
+require 'procedures/backup/snapshot/mount_base'
+module Procedures::Backup
+  module Snapshot
+    class MountForemanDB < MountBase
+      metadata do
+        description 'Create and mount snapshot of Foreman DB'
+        tags :backup
+        label :backup_snapshot_mount_foreman_db
+        for_feature :foreman_database
+        preparation_steps { Checks::Foreman::DBUp.new unless feature(:foreman_database).local? }
+        MountBase.common_params(self)
+        param :backup_dir, 'Directory where to backup to'
+      end
+
+      def run
+        if feature(:foreman_database).local?
+          mount_snapshot
+        else
+          dump_foreman
+        end
+      end
+
+      private
+
+      def dump_foreman
+        puts 'LV snapshots are not supported for remote databases. Doing postgres dump instead... '
+        with_spinner('Getting Foreman DB dump') do
+          feature(:foreman_database).dump_db(File.join(@backup_dir, 'foreman.dump'))
+        end
+      end
+
+      def mount_snapshot
+        if Dir.entries(mount_location('pgsql')).empty?
```

Comment:
same as mount_candlepin_db.rb

Reply:
This line checks if the snapshot was already mounted by mount_candlepin_db